### PR TITLE
Redundant lazy members test

### PR DESCRIPTION
**Pull request by ara4n**
Tests https://github.com/matrix-org/synapse/pull/3331

(also includes https://github.com/matrix-org/sytest/pull/445)

### CODE REVIEW

**Comment by ara4n** [comment]
@matrixbot retest this please

File: tests/31sync/15lazy-members.pl [hunk 1]
```diff
@@ -200,6 +200,167 @@
...
+      # Bob sends 10 events into it
+      # Charlie sends 5 events into it
+      # Alice syncs with a filter on the last 10 events, and lazy loaded members
+      # Alice should see only Bob and Charlie in the membership list.
+      # Bob sends 1 more event
+      # Charlie sends 1 more event
+      # Alice syncs again; she should not see any membership events as
+      # the redundant ones for Bob and Charlie are removed.
+
+      # TODO: speed up time and check that if we wait an hour then the server's
```
Comment (richvdh) [inline tests/31sync/15lazy-members.pl]: isn't this an implementation issue? servers can keep the cache for as long or as short as they want?

**Comment by richvdh** [comment]
> looks plausible. I'd like to see more validation of the filter params.

or rather, I'd like to see synapse doing more validation, and I'd like to see sytest doing more validation of that validation.

but I'm not going to insist on it for now.

**Comment by ara4n** [comment]
done

**Pull request merged**